instructor/common.py

```diff
@@ -667,6 +669,81 @@ def western_blot_band_size(request):
         context_instance=RequestContext(request))
 
 
+def is_float(txt):
+    try:
+        float(txt)
+        return True
+    except ValueError:
+        return False
+
+@login_required
+def western_blot_band_intensity(request):
+    pk = request.session['assignment_id']
+    assignment = models.Assignment.objects.get(id=pk)
+    wb, created = models.WesternBlot.objects.get_or_create(assignment=assignment)
+    BandsFormset = modelformset_factory(
+        models.WesternBlotBands,
+        extra=0,
+        exclude=['strain_protocol', 'antibody', 'weight', 'lysate_type']
+    )
+    # Create Bands
+    antibodies = models.WesternBlotAntibody.objects.filter(western_blot=wb)
+    strain_treatments = models.StrainTreatment.objects.filter(assignment=assignment)
+    weights_by_type = ['wc_weight', 'nuc_weight', 'cyto_weight']
+    for antibody in antibodies:
+        for strain_treatment in strain_treatments:
+            for index, field in enumerate(weights_by_type):
+                weight_str = getattr(antibody, field)
+                weights = [float(s) for s in weight_str.split(',')
+                           if is_float(s) and float(s) >= 0]
+                for weight in weights:
+                    models.WesternBlotBands.objects.get_or_create(
+                        strain_protocol=strain_treatment,
+                        antibody=antibody,
+                        weight=weight,
+                        lysate_type=field.split('_')[0]
+                    )
+    if request.method == 'POST':
+        formset = BandsFormset(request.POST)
+        if formset.is_valid():
+            entries = formset.save(commit=False)
+            for entry in entries:
+                entry.save()
+    else:
+        formset = BandsFormset(queryset=models.WesternBlotBands.objects.filter(antibody__western_blot=wb))
+
+    formset_group = []
+    for antibody in antibodies:
+        form_list = []
```

Comment:
This can be rewritten as a list comprehension: 

    form_list = [form for form in form_set 
     if form.instance.antibody.primary == antibody.primary]